ProcessMaker/Http/Controllers/TaskController.php

```diff
@@ -104,6 +106,20 @@ public function edit(ProcessRequestToken $task, string $preview = '')
                 ]);
             }
 
+            $existingView = UserResourceView::where('user_id',Auth::user()->id)
+                ->where('viewable_type', ProcessRequest::class)
+                ->where('viewable_id', $task->id)
+                ->first();
+
+            if (!$existingView) {
+                $userResourceView = new UserResourceView();
+                $userResourceView->user_id = Auth::user()->id;
+                $userResourceView->viewable_type = ProcessRequestToken::class;
+                $userResourceView->viewable_id = $task->id;
+                $userResourceView->save();
+
+            }
+            
```

Comment:
@gustavobascope please move this logic to a new static method in the UserResourceView class so it can be called like `UserResourceView::setViewed(Auth::user, $task)`. You can probably use `self::firstOrCreate(...)` in the new static method. Same for RequestController

Reply:
Observation completed